src/librawspeed/common/Point.h

```diff
@@ -77,23 +77,21 @@ class iPoint2D {
   area_type __attribute__((pure)) area() const {
     using signed_area = std::make_signed<area_type>::type;
 
-    if (x >= 0 && y >= 0)
-      return static_cast<area_type>(x) * static_cast<area_type>(y);
-    if (x >= 0 && y < 0)
-      return static_cast<area_type>(x) * (-1 * static_cast<signed_area>(y));
-    if (y >= 0 && x < 0)
-      return static_cast<area_type>(y) * (-1 * static_cast<signed_area>(x));
+    area_type x_abs = x < 0 ? std::abs(static_cast<signed_area>(x)) : x;
+    area_type y_abs = y < 0 ? std::abs(static_cast<signed_area>(y)) : y;
```

Comment:
Right.
But then let's take it all the way, and just do
```diff
-    area_type x_abs = x < 0 ? std::abs(static_cast<signed_area>(x)) : x;
-    area_type y_abs = y < 0 ? std::abs(static_cast<signed_area>(y)) : y;
+    area_type x_abs = std::abs(static_cast<signed_area>(x));
+    area_type y_abs = std::abs(static_cast<signed_area>(y));
```
?

Reply:
Fixed.